<a href="https://colab.research.google.com/github/hermine-dlmr/qm2-group0/blob/main/iCovid_19ue.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving data.3.csv to data.3.csv


In [ ]:
import pandas as pd
import plotly.express as px

data = pd.read_csv('data.3.csv', delimiter = ';' , index_col= 0, decimal=",", parse_dates=["Date"])

In [ ]:
print(data.to_csv(columns=['Date', 'Unemployment Rate [%]'], sep='\t', index=False))

Date	Unemployment Rate [%]
2000-01-01	4.0
2000-02-01	4.1
2000-03-01	4.0
2000-04-01	3.8
2000-05-01	4.0
2000-06-01	4.0
2000-07-01	4.0
2000-08-01	4.1
2000-09-01	3.9
2000-10-01	3.9
2000-11-01	3.9
2000-12-01	3.9
2001-01-01	4.2
2001-02-01	4.2
2001-03-01	4.3
2001-04-01	4.4
2001-05-01	4.3
2001-06-01	4.5
2001-07-01	4.6
2001-08-01	4.9
2001-09-01	5.0
2001-10-01	5.3
2001-11-01	5.5
2001-12-01	5.7
2002-01-01	5.7
2002-02-01	5.7
2002-03-01	5.7
2002-04-01	5.9
2002-05-01	5.8
2002-06-01	5.8
2002-07-01	5.8
2002-08-01	5.7
2002-09-01	5.7
2002-10-01	5.7
2002-11-01	5.9
2002-12-01	6.0
2003-01-01	5.8
2003-02-01	5.9
2003-03-01	5.9
2003-04-01	6.0
2003-05-01	6.1
2003-06-01	6.3
2003-07-01	6.2
2003-08-01	6.1
2003-09-01	6.1
2003-10-01	6.0
2003-11-01	5.8
2003-12-01	5.7
2004-01-01	5.7
2004-02-01	5.6
2004-03-01	5.8
2004-04-01	5.6
2004-05-01	5.6
2004-06-01	5.6
2004-07-01	5.5
2004-08-01	5.4
2004-09-01	5.4
2004-10-01	5.5
2004-11-01	5.4
2004-12-01	5.4
2005-01-01	5.3
2005-02-01	5.4
2005-03-01	5.2
2005-04-01	5.2
2005-05-01	5.

In [ ]:
graph = px.line(data, 
                x = "Date",
                y = "Unemployment Rate [%]", 
                title="United States Unemployment Rate [2000 - 2020]", 
                template="plotly_white",
                height=500,
                width=750)
graph.show()

!pip install chart_studio
import chart_studio
username = 'jo1708' # your username
api_key = 'CJnrqb5RtHQMiYQ7x0IO' # your api key - go to profile > settings > regenerate key
chart_studio.tools.set_credentials_file(username=username, api_key=api_key)
import chart_studio.plotly as py
py.plot(graph, filename = 'The unemployment Rate ', auto_open=True)

'https://plotly.com/~jo1708/53/'

This graph shows that when US states were going into lockdown (varying from March 22nd to April 12th), the unemployment rate climbed to its highest percentage in the time period between 2000 to 2020, namely 14.7 % in April 2020. This shows how Covid-19 has affected employment across the United States. 

data taken from: https://data.bls.gov/timeseries/LNS14000000?years_option=all_years

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving theUE_19:20.csv to theUE_19:20.csv


In [ ]:
data_bar = pd.read_csv('UE_19:20.csv')#, delimiter = ';' , index_col= 0, decimal=",")
data_bar.head()

Unnamed: 0                Ethnicity  Year Unemployment Rate
0           0                    White  2019               3,4
1           1  African-American/ Black  2019               5,6
2           2         Hispanic/ Latino  2019               4,2
3           3                    Asian  2019               2,8
4           4                    White  2020               7,9

In [ ]:

data_bar["Year"] = data_bar["Year"].astype(str)
bar_chart = px.bar(data_bar,
                   x="Ethnicity", 
                   y="Unemployment Rate",
                   color="Year",
                   title="Unemployment Rate [%] by ethnicity [3rd Quarter 2019 vs. 3rd Quarter 2020]",
                   template="plotly_white")
bar_chart.show()

!pip install chart_studio
import chart_studio
username = 'jo1708' # your username
api_key = 'CJnrqb5RtHQMiYQ7x0IO' # your api key - go to profile > settings > regenerate key
chart_studio.tools.set_credentials_file(username=username, api_key=api_key)
import chart_studio.plotly as py
py.plot(bar_chart, filename = 'Unempoyment and ethnicity', auto_open=True)

NameError: ignored

This bar chart shows the unemployment rate by race or ethnicity for the third quarter of 2019 vs the third quarter of 2020 in order to illustrate how the Covid-19 pandemic and the subsequent containment-measures have affected various demographic groups differently in regards to unemployment. 

For example, whilst White Americans have seen an increase by 4.5 %, unemployment amongst African-Americans or Black Americans has increased by 7.6%. The highest increase, namely 7.8 % can be witnessed amongst Asian Americans. This could be attributed to the fact that the employment sectors that were affected most by the pandemic are hospitality and leisure. In the US the highest percentage of workers that are occupied in hospitality or leisure (service sector) are African-American and Hispanic with 24%. (It has to be mentioned that the high increase in Asian American unemployment cannot be explained properly by the Covid-19 pandemic, as 54% of Asian Americans work in the Management industry which has only seen a decline of 5%.) 

data taken from: https://www.bls.gov/web/empsit/cpsee_e16.htm
https://www.stlouisfed.org/publications/regional-economist/third-quarter-2020/jobs-hit-hardest-covid-19
https://www.bls.gov/opub/reports/race-and-ethnicity/2018/home.htm#:~:text=The%20employment%E2%80%93population%20ratio%20was,and%2063.2%20percent%20for%20Hispanics.
